tests/test_utils_transformers.py

```diff
@@ -20,18 +20,26 @@ def test_basic_func(self, X):
         "X_new", [np.array([1, 1]).reshape(-1, 1), np.array([[[1], [1]]])]
     )
     def test_invalid_input_transform(self, X_new):
-        tf = Ensure2DTransformer()
-        X = np.array([1, 1])
         # Fit with 1D then pass 2D or 3D
-        # Raises ValueError
-        tf.fit(X)
-        with pytest.raises(ValueError, match="Expected a 1D array"):
+        X = np.random.uniform(size=100)
+        tf = Ensure2DTransformer().fit(X)
+
+        if X.ndim == 1:
+            with pytest.raises(ValueError, match="Expected a 1D array"):
```

Comment:
Should this error be raised? In the case this error is raised, there's only one meaningful dimension in `X_new`. Is this error raised when `X_new` has two dimensions and only one dimension is meaningful?

Reply:
I think you originally introduced the error. I noticed it wasn't being tested (from coverage) and I thought I was fixing the test with this change. I could be wrong. I believe the idea is that if the user passes `X` as a column vector (specifically a column vector i.e. `(n_samples,)` and not `(1, n_samples, 1)`) to `fit` and then passes a >1D  `X` (ex: `(n_samples, 1)` or `(1, n_samples, 1)`) to `transform` this error should be raised. Does that make sense?

Reply:
It looks like it was [here](https://github.com/adriangb/scikeras/blob/32a878cf8bc56cdf84d9679930cd7b329ad07d29/scikeras/utils/transformers.py#L17-L22).

I don't fully understand what the expected behavior is with `X.shape == (1, 400, 1)` since it that would result in `self.should_transform_ = False` from `fit` and hence this branch in `transform` would never be run. Am I missing something?

Reply:
To me it makes sense as it is now (without any mention of `X.shape == (1, 400, 1)`). I'll leave it as it is now then.